packages/wp-comments/src/index.ts

```diff
@@ -0,0 +1,153 @@
+import WpComments from "../types";
+
+const wpComments: WpComments = {
+  name: "@frontity/wp-comments",
+  state: {
+    comments: {
+      forms: {},
+    },
+  },
+  actions: {
+    comments: {
+      submit: ({ state, actions }) => async (postId, comment) => {
+        // Check first if the connected source is a WP.com site.
+        if (state.source.isWpCom) {
+          return warn(
+            "Sending comments to a WordPress.com site is not supported yet."
+          );
+        }
+
+        // Return if a submission is pending.
+        if (state.comments.forms[postId]?.submitted?.isPending) {
+          return warn(
+            "You cannot submit a comment to the same post if another is already pending."
+          );
+        }
+
+        // Update fields for this form.
+        // This line inits the form if it wasn't yet.
+        actions.comments.updateFields(postId, comment || {});
+
+        // Get fields from the corresponding form.
+        const form = state.comments.forms[postId];
+        const { fields } = form;
+
+        // Reset the `submitted` object.
+        form.submitted = {
+          isError: false,
+          errorMessage: "",
+          isPending: true,
+          isUnapproved: false,
+          isApproved: false,
+          timestamp: Date.now(),
+          ...fields,
+        };
+
+        // Generate form content.
+        const body = new URLSearchParams();
+        body.set("comment", fields.comment);
+        body.set("author", fields.author);
+        body.set("email", fields.email);
+        body.set("url", fields.url);
+        body.set("comment_parent", fields.parent.toString());
+        body.set("comment_post_ID", postId.toString());
+
+        // Generate endpoint URL.
+        const commentsPost = state.source.api.replace(
+          /\/wp-json\/?$/,
+          "/wp-comments-post.php"
+        );
+
+        // Send a POST request.
+        let response: Response;
+        try {
+          response = await fetch(commentsPost, {
+            method: "POST",
+            headers: { "Content-Type": "application/x-www-form-urlencoded" },
+            redirect: "manual",
+            body,
+          });
+        } catch (e) {
+          // Network error.
+          form.submitted.isPending = false;
+          form.submitted.isError = true;
+          form.submitted.errorMessage = "Network error";
+          return;
+        }
+
+        // 200 OK - The request has failed because the post ID is invalid, name
+        //          or email are missing or email has an invalid format.
+        if (response.status === 200) {
+          // Get first the body content.
+          const body = await response.text();
+
+          // Set error properties
+          form.submitted.isPending = false;
+          form.submitted.isError = true;
+
+          // If the body is empty `postId` is invalid;
+          // `author` or `email` are invalid otherwise.
+          form.submitted.errorMessage = !body
```

Comment:
This is more of a side question, are there any plans on supporting internationalization? Having something like `__()` to localize static strings

This is something that can be handled at the theme level but there are probably a bunch of hardcoded messages across the codebase and having a standard way of handling this kind of thing would be useful.

Reply:
Yeah, it would be useful!

It is not something in our roadmap right now but you can create a topic in the [Feature Discussions](https://community.frontity.org/c/feature-discussions/33) section for that. :slightly_smiling_face:

Reply:
Oh, yes, sorry. A standard internationalization is on the roadmap. Just no FD yet. I have some ideas though :)